src/types/basic.ts

```diff
@@ -182,5 +182,83 @@ export const DeviceName = new t.Type<string, string>(
 		),
 	t.identity,
 );
-
 export type DeviceName = t.TypeOf<typeof DeviceName>;
+
+/**
+ * A string boolean, one of ['true', 'false', '1', '0', 'on', 'off'].
+ * Decodes to a boolean.
+ */
+
+export const BooleanFromString = new t.Type<boolean, string, unknown>(
+	'BooleanFromString',
+	t.boolean.is,
+	(i, c) =>
+		pipe(
+			t.string.validate(i, c),
+			chain((s) => {
+				if (['true', '1', 'on'].includes(s)) {
+					return t.success(true);
+				}
+
+				if (['false', '0', 'off'].includes(s)) {
+					return t.success(false);
+				}
+
+				return t.failure(i, c, 'must be a valid string boolean');
```

Comment:
Is this too strict? Would it suffice with making anything not in `['true', '1', 'on']` as false?